version: '3'

networks:
  local:

services:
  postgres:
    container_name: postgres_container
    image: postgres:10.12
    environment:
      POSTGRES_USER: UserLocal
      POSTGRES_PASSWORD: UserLocal
      POSTGRES_DB: LocalDB
    ports:
      - "5432:5432"
    restart: unless-stopped
    networks:
      local:
        aliases:
          - postgres_backend_api

  z1socialideas:
    tty: true
    build:
      context: .
      dockerfile: Dockerfile
      args:
        - pipenv_dev=1
    environment:
      - PYTHONDONTWRITEBYTECODE=1
      - PIPENV_DOTENV_LOCATION=/tmp/.env
    working_dir: /code
    volumes:
      - ./Z1DjangoGraphql:/code
      - ./.env:/tmp/.env
    command: bash -c "
      pipenv run python manage.py migrate &&
      pipenv run python manage.py loaddata flags &&
      pipenv run gunicorn --name app --workers 2 --log-level debug --error-logfile /tmp/gunicorn_err.log --reload --bind 0.0.0.0:8000 backend_api.wsgi:application
      "
    ports:
      - 8000:8000
    networks:
      local:

  worker:
    build:
      context: .
      dockerfile: Dockerfile
    environment:
      - PYTHONDONTWRITEBYTECODE=1
      - PIPENV_DOTENV_LOCATION=/tmp/.env
    working_dir: /code
    volumes:
      - ./Z1DjangoGraphql:/code:ro
      - ./.env:/tmp/.env:ro
    command: bash -c "
      pipenv run celery -A Z1DjangoGraphql.settings worker -B -c 4 --loglevel=DEBUG
     "
    networks:
      local:
